python/social_bot/teacher.py

```diff
@@ -122,13 +123,17 @@ def reset(self, agent, world):
     def _get_current_task(self):
         if self._current_task is None:
             tid = random.randint(0, len(self._tasks) - 1)
+            self._current_tid = tid
             self._current_task = self._tasks[tid].run()
             self._current_reward_weight = self._tasks[tid].reward_weight
             # This send will cause self._current_task to execute until the first
             # yield. We ignore the first yielded value.
             self._current_task.send(None)
         return self._current_task
 
+    def get_current_task_non_generator(self):
```

Comment:
what does "_non_generator" mean here?

Reply:
default get_current_task returns a python generator function, instead of the task object.  Maybe we should rename the default one as well?

Reply:
Seems that this function is not used anywhere